pkg/dataobj/internal/sections/logs/iter.go

```diff
@@ -93,13 +95,16 @@ func IterSection(ctx context.Context, dec encoding.LogsDecoder, section *filemd.
 // Decode decodes a record from a [dataset.Row], using the provided columns to
 // determine the column type. The list of columns must match the columns used
 // to create the row.
-func Decode(columns []*logsmd.ColumnDesc, row dataset.Row) (Record, error) {
-	record := Record{
-		// Preallocate metadata to exact number of metadata columns to avoid
-		// oversizing.
-		Metadata: make(labels.Labels, 0, metadataColumns(columns)),
+func Decode(columns []*logsmd.ColumnDesc, row dataset.Row, record *Record) error {
+	if record.Line == nil {
+		record.Line = make([]byte, 0)
 	}
```

Comment:
I'm pretty sure initializing a nil slice to a slice of length 0 is a no-op*, so we could just omit this line here

*Because Go lets you append to nil slices, and I don't think it allocates anything if the length is zero (which also implies the capacity is zero)

Reply:
Hmm, I added this for a reason, because copy wasn't working as expected, but it seems my later iterations on slicegrow.Copy does the right thing so its no longer needed. Removed.